File: apps/web/lib/auth/module-access.ts
export type Module = 'dashboard' | 'financial' | 'warehouse' | 'admin';

export const MODULE_ACCESS: Record<Module, string[]> = {
  dashboard: ['ADMIN', 'MANAGER', 'ACCOUNTING', 'WAREHOUSE'],
  financial: ['ADMIN', 'MANAGER', 'ACCOUNTING'],
  warehouse: ['ADMIN', 'MANAGER', 'WAREHOUSE'],
  admin: ['ADMIN'],
};

export function canAccessModule(module: Module, role: string | undefined | null): boolean {
  if (!role) return false;
  return MODULE_ACCESS[module].includes(role);
}

File: apps/api/Domain/Constants/ModuleAccess.cs
using System.Linq;

namespace Pinoles.Api.Domain.Constants;

/// <summary>Single source of truth for module → allowed-roles mapping. Mirrors the
/// authorization policies in Program.cs and the frontend MODULE_ACCESS map.</summary>
public static class ModuleAccess
{
    public static readonly string[] Dashboard = { UserRoles.Admin, UserRoles.Manager, UserRoles.Accounting, UserRoles.Warehouse };
    public static readonly string[] Financial = { UserRoles.Admin, UserRoles.Manager, UserRoles.Accounting };
    public static readonly string[] Warehouse = { UserRoles.Admin, UserRoles.Manager, UserRoles.Warehouse };
    public static readonly string[] Admin = { UserRoles.Admin };

    public static bool CanAccess(string[] moduleRoles, string role) => moduleRoles.Contains(role);
}
